### PR TITLE
Cleannig purchase module a bit

### PR DESCRIPTION
**Pull request by DedrickEnc**
This set of commits clean the purchase order code a bit and introduces the ```bhSupplierSelect``` component and update the e2e test.

### CODE REVIEW

File: client/src/modules/purchases/create/PurchaseOrderForm.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ function PurchaseOrderFormService(Inventory, AppCache, Store, Pool, PurchaseOrde
     this.inventory = new Pool({ identifier: 'uuid', data : [] });
 
     // set up the inventory
-    Inventory.read()
+    Inventory.read(null, { locked : 0 })
```
Comment (jniles) [inline client/src/modules/purchases/create/PurchaseOrderForm.js:43]: :dancer:

**Review by jniles — COMMENTED** [review]
While you are at it, would you mind changing `$translate` filters to directives?

**Comment by DedrickEnc** [comment]
@jniles

**Review by jniles — APPROVED** [review]
LGTM!

bors r+

**Pull request merged**